ui/apps/platform/src/Components/CompoundSearchFilter/components/SearchFilterChips.tsx

```diff
@@ -51,47 +55,51 @@ export function makeFilterChipDescriptors(
 ): FilterChipGroupDescriptor[] {
     const filterChipDescriptors = config.flatMap(
         ({ attributes = [] }: CompoundSearchFilterEntity) =>
-            attributes.map((attribute) => {
-                const baseConfig = {
-                    displayName: attribute.filterChipLabel,
-                    searchFilterName: attribute.searchTerm,
-                };
-
-                if (isSelectType(attribute)) {
-                    const options = hasGroupedSelectOptions(attribute.inputProps)
-                        ? attribute.inputProps.groupOptions.flatMap((group) => group.options)
-                        : attribute.inputProps.options;
-                    return {
-                        ...baseConfig,
-                        render: (filter: string) => {
-                            const option = options.find((option) => option.value === filter);
-                            return <FilterChip name={option?.label || 'N/A'} />;
-                        },
-                    };
-                }
-
-                if (attribute.inputType === 'condition-text') {
-                    return {
-                        ...baseConfig,
-                        render: (filter: string) => {
-                            return (
-                                <FilterChip
-                                    name={convertFromInternalToExternalConditionText(
-                                        attribute.inputProps,
-                                        filter
-                                    )}
-                                />
-                            );
-                        },
-                    };
-                }
-
-                return baseConfig;
-            })
+            attributes.map(makeFilterChipDescriptorFromAttribute)
     );
     return filterChipDescriptors;
 }
 
+export function makeFilterChipDescriptorFromAttribute(
+    attribute: CompoundSearchFilterAttribute
+): FilterChipGroupDescriptor {
+    const baseConfig = {
+        displayName: attribute.filterChipLabel,
+        searchFilterName: attribute.searchTerm,
+    };
+
+    if (isSelectType(attribute)) {
```

Comment:
Just an observation, seeing this util next to the explicit `if (attribute.inputType === 'condition-text') {` below made me think the `isSelectType` was doing something more than `inputType === 'select'`. I was let down.